reps/2022-08-31-actor-affinity-apis.md

```diff
@@ -0,0 +1,538 @@
+
+## Summary
+### General Motivation
+
+Provides a set of lightweight actor affinity and anti-affinity scheduling interfaces.
+Replacing the heavy PG interface to implement collocate or spread actors.
+
+* Affinity
+  * Co-locate the actors in the same node.
+  * Co-locate the actors in the same batch of nodes, like nodes in the same zones
+* Anti-affinity
+  * Spread the actors of a service across nodes and/or availability zones, e.g. to reduce correlated failures.
+  * Give a actor "exclusive" access to a node to guarantee resource isolation 
+  * Spread the actors of different services that will affect each other on different nodes.
+
+
+### Should this change be within `ray` or outside?
+
+Yes, this will be a complement to ray core's ability to flexibly schedule actors/tasks.
+
+## Stewardship
+### Required Reviewers
+ 
+@wumuzi520 SenlinZhu @Chong Li  @scv119 (Chen Shen) @jjyao (Jiajun Yao) 
+### Shepherd of the Proposal (should be a senior committer)
+
+
+## Design and Architecture
+
+### Brief idea
+
+1. Tag the Actor with key-value labels first
+2. Then affinity or anti-affinity scheduling to the actors of the specified labels.
+
+![ActorAffinityScheduling](https://user-images.githubusercontent.com/11072802/188054945-48d980ed-2973-46e7-bf46-d908ecf93b31.jpg)
```

Comment:
Are there any examples of existing systems that have this ActorAffinity concept? Or was there any existing inspiration for this?

Reply:
This ActorAffinity is similar to podAffinity/NodeAffinity of k8s.